_security/configuration/opensearch-keystore.md

```diff
@@ -0,0 +1,70 @@
+---
+layout: default
+title: OpenSearch keystore
+parent: Configuration
+nav_order: 50
+---
+
+# OpenSearch Keystore
+
+`opensearch-keystore` is a utility script for managing the OpenSearch keystore. This keystore provides a secure method to store sensitive information, such as passwords and keys, used in the OpenSearch cluster. The script allows users to create, list, add, and remove settings securely.
+
+## Usage
+
+```
+opensearch-keystore [command] [options]
+```
+
+### Commands
+
+- `create`: Initializes a new keystore. If a keystore already exists, this command will overwrite it.
+- `list`: Lists all settings in the keystore.
```

Comment:
Is there a place I can find a list of settings I can to the keystore?

Reply:
The only place you can see a full list is in the source code [here](https://github.com/opensearch-project/OpenSearch/blob/main/distribution/tools/keystore-cli/src/main/java/org/opensearch/common/settings/KeyStoreCli.java#L43-L53). This [article](https://opster.com/guides/opensearch/opensearch-security/opensearch-keystore/) from Opster is also a good reference for usage of the opensearch-keystore tool.